processing/src/main/java/org/apache/druid/query/context/ConcurrentResponseContext.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.context;
+
+import org.apache.druid.guice.annotations.PublicApi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The implementation of {@link ResponseContext} with a {@link ConcurrentHashMap} as a delegate
```

Comment:
Could you add to the javadoc of this and `DefaultResponseContext` when each is used?

Reply:
Unfortunately, I don't fully understand when each is used since it's not obvious from the codebase. The only thing I found about ConcurrentHashMap usage is `Fix a race condition in CoordinatorRuleManager` by @leventov. So from my point of view concurrent map is not an architecture-driven solution but more like a bug-driven solution